cupy/cuda/pinned_memory.pyx

```diff
@@ -330,6 +330,7 @@ cdef class PinnedMemoryPool:
 
     cpdef free_all_blocks(self):
         """Release free all blocks."""
+        _watcher.check_and_release()
```

Comment:
Q: it seems this method also needs to acquire the rlock. How about calling `_check_and_release_without_lock()` in the try block below?

Reply:
These two locks are the different things. Both Watcher and PinnedMemoryPointer has each lock instance to protect their attributes.

Reply:
Ah, I see! Thanks for pointing it out.